janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/MultiVertexCentricQueryBuilder.java

```diff
@@ -202,4 +203,19 @@ public Map<JanusGraphVertex, VertexList> vertexIds() {
         return execute(RelationCategory.EDGE, new VertexIdConstructor());
     }
 
+    @Override
+    public Integer drop() {
+        Map<JanusGraphVertex, Iterable<JanusGraphRelation>> relations = this.noPartitionRestriction().relations();
```

Comment:
Here it is possible to reduce queries by half

```
Subject: [PATCH]
---
Index: janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java b/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java
--- a/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java	(revision 42faab99d80a68a49454102e0e30e2d1f2cba57b)
+++ b/janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/BasicVertexCentricQueryBuilder.java	(revision 6c7ce256fa20fa13f4612ab84eda0d05853902f0)
@@ -30,6 +30,7 @@
 import org.janusgraph.core.schema.SchemaStatus;
 import org.janusgraph.diskstorage.keycolumnvalue.SliceQuery;
 import org.janusgraph.graphdb.database.EdgeSerializer;
+import org.janusgraph.graphdb.database.idhandling.IDHandler;
 import org.janusgraph.graphdb.internal.ElementLifeCycle;
 import org.janusgraph.graphdb.internal.InternalRelationType;
 import org.janusgraph.graphdb.internal.InternalVertex;
@@ -101,6 +102,7 @@
      * Whether to query for system relations only
      */
     private boolean querySystem = false;
+    private boolean all = false;
     /**
      Whether to query only for persisted edges, i.e. ignore any modifications to the vertex made in this transaction.
      This is achieved by using the {@link SimpleVertexQueryProcessor} for execution.
@@ -165,6 +167,11 @@
         return getThis();
     }
 
+    public Q all() {
+        this.all = true;
+        return getThis();
+    }
+
     /**
      * Calling this method will cause this query to only included loaded (i.e. unmodified) relations in the
      * result set.
@@ -467,10 +474,22 @@
         return new BaseVertexCentricQuery(condition, Direction.OUT, queries, OrderList.NO_ORDER, Query.NO_LIMIT);
     }
 
+    private BaseVertexCentricQuery getFullQuery() {
+        SliceQuery fullQuery = new SliceQuery(
+            IDHandler.getBounds(RelationCategory.RELATION, true)[0],
+            IDHandler.getBounds(RelationCategory.RELATION,false)[1]);
+
+        List<BackendQueryHolder<SliceQuery>> queries = Collections.singletonList(new BackendQueryHolder<>(fullQuery, true, true));
+        return new BaseVertexCentricQuery(RelationCategory.RELATION, Direction.BOTH, queries, OrderList.NO_ORDER, Query.NO_LIMIT);
+    }
+
     protected BaseVertexCentricQuery constructQueryWithoutProfile(RelationCategory returnType) {
         assert returnType != null;
         Preconditions.checkArgument(adjacentVertex==null || returnType == RelationCategory.EDGE,
                 "Vertex constraints only apply to edges");
+        if (all) {
+            return getFullQuery();
+        }
         if (isVertexLabelType()) {
             return queryForLabel();
         }
Index: janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/AbstractVertex.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/AbstractVertex.java b/janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/AbstractVertex.java
--- a/janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/AbstractVertex.java	(revision 42faab99d80a68a49454102e0e30e2d1f2cba57b)
+++ b/janusgraph-core/src/main/java/org/janusgraph/graphdb/vertices/AbstractVertex.java	(revision 6c7ce256fa20fa13f4612ab84eda0d05853902f0)
@@ -103,13 +103,7 @@
     public synchronized void remove() {
         verifyAccess();
 //        if (isRemoved()) return; //Remove() is idempotent
-        Iterator<JanusGraphRelation> iterator = it().query().noPartitionRestriction().relations().iterator();
-        while (iterator.hasNext()) {
-            iterator.next();
-            iterator.remove();
-        }
-        //Remove all system types on the vertex
-        for (JanusGraphRelation r : it().query().noPartitionRestriction().system().relations()) {
+        for (JanusGraphRelation r : it().query().noPartitionRestriction().all().relations()) {
             r.remove();
         }
     }
```

Reply:
nice!

Reply:
i have tested this approach, it doesn't seem to return all relations, it returns only half of it

Reply:
never mind, i had a wrong check. The code works as expected.
I have integrated this change into mine, thanks again @mad !

Reply:
With a new query, this part `AbstractVertex.remove()` can be done in one pass

Reply:
Updated